cli: Fix `persist-nic-names` on bond port

cathay4t opened the pull request:
When interface been attached to bond, all bond ports might holding the
same MAC address of bond.

Invoking `persist-nic-names` on bond port which does not have permanence
MAC address will cause nmstatectl using this incorrect MAC to persistent
the NIC name.

To fix this problem, for bond port, we find its permanence MAC address
first, if failed, we use MAC address stored in netlink
`IFLA_BOND_PORT_PERM_HWADDR` message. The current MAC address will never
been use for bond port.

In order to retrieve the `IFLA_BOND_PORT_PERM_HWADDR`, we introduce
dependency on `nispor` to get such information. Considering `nmstate` crate
also depend on `nispor`, we are not expanding our dependency chain.

cgwalters commented on rust/src/cli/persist_nic.rs:
```diff
@@ -417,3 +438,48 @@ fn has_any_kargs(kargs: &[&str]) -> bool {
+    }
+
+    let mut filter = nispor::NetStateFilter::minimum();
+    let mut iface_fitler = nispor::NetStateIfaceFilter::minimum();
```
s/fitler/filter/ ?

cathay4t replied:
Fixed. Thanks!

cathay4t commented:
@cgwalters Is the patch OK to you now?

cgwalters reviewed (approved):
I can only give this a superficial ACK

liangwen12year commented on rust/src/cli/persist_nic.rs:
```diff
@@ -417,3 +438,48 @@ fn has_any_kargs(kargs: &[&str]) -> bool {
+// 2. MAC address stored in `IFLA_BOND_PORT_PERM_HWADDR`
+fn get_bond_port_mac(iface: &Interface) -> Option<String> {
+    if let Some(mac) = iface.base_iface().permanent_mac_address.as_deref() {
+        return Some(mac.to_string());
```
A nit, I noticed that when using the MAC address stored in `IFLA_BOND_PORT_PERM_HWADDR`, we add a log, should we also add a log here before return ?

liangwen12year reviewed (commented):
In the commit message, 

```
Invoking `persist-nic-names` on bond port which does not have permanence
MAC address will cause nmstatectl using this incorrect MAC to persistent
the NIC name.
```

Replace `permanence` with `permanent` ? 
Replace ` persistent the NIC name.` with the ` as the persistent NIC name (ifname={ifname}:{mac})` ?

cathay4t committed:
cli: Fix `persist-nic-names` on bond port

When interface been attached to bond, all bond ports might holding the
same MAC address of bond.

Invoking `persist-nic-names` on bond port which does not have permanent
MAC address will cause nmstatectl using this incorrect MAC to pin the
NIC name.

To fix this problem, for bond port, we find its permanence MAC address
first, if failed, we use MAC address stored in netlink
`IFLA_BOND_PORT_PERM_HWADDR` message. The current MAC address will never
been use for bond port.

In order to retrieve the `IFLA_BOND_PORT_PERM_HWADDR`, we introduce
dependency on nispor to get such information. Considering nmstate crate
also depend on nispor, we are not expanding our dependency chain.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
Replaced permanence with permanent.

Changed from `to persistent the NIC name` to `to pin the NIC name`.
